include/boost/config/compiler/nvcc.hpp

```diff
@@ -11,6 +11,15 @@
 #  define BOOST_COMPILER "NVIDIA CUDA C++ Compiler"
 #endif
 
+#if defined(__CUDACC_VER_MAJOR__) && defined(__CUDACC_VER_MINOR__) && defined(__CUDACC_VER_BUILD__)
+#  define BOOST_CUDA_VERSION __CUDACC_VER_MAJOR__ * 10000 + __CUDACC_VER_MINOR__ * 100 + __CUDACC_VER_BUILD__
```

Comment:
@jzmaddock careful, even CUDA 9 RC2 is already above 100 builds in `__CUDACC_VER_BUILD__`.

In my bug report I suggested to either increase for CUDA 9 build by a full order (*10) or to crop the build to two significant ints from the left during deprecation.